google/cloud/storage/client.h

```diff
@@ -160,28 +161,28 @@ struct ClientImplDetails;
  * In addition, the @ref index "main page" contains examples using `StatusOr<T>`
  * to handle errors.
  *
- * @par Optional Parameters
- * Most of the member functions in this class can receive optional parameters
- * to modify their behavior. For example, the default when reading multi-version
- * objects is to retrieve the latest version:
+ * @par Optional Request Options
+ * Most of the member functions in this class can receive optional request
+ * options. For example, the default when reading multi-version objects is to
+ * retrieve the latest version:
  *
  * @code
  * auto stream = gcs.ReadObject("my-bucket", "my-object");
  * @endcode
  *
- * Some applications may want to retrieve specific versions, in this case
- * just provide the `Generation` option:
+ * Some applications may want to retrieve specific versions. In this case
+ * just provide the `Generation` request option:
  *
  * @code
  * auto stream = gcs.ReadObject(
  *     "my-bucket", "my-object", gcs::Generation(generation));
  * @endcode
  *
- * Each function documents the types accepted as optional parameters. These
- * options can be specified in any order. Specifying an option that is not
- * applicable to a member function results in a compile-time error.
+ * Each function documents the types accepted as optional request options. These
+ * parameters can be specified in any order. Specifying a request option that is
+ * not applicable to a member function results in a compile-time error.
```

Comment:
Is this true?  If so, I'm not sure it is a feature.

Reply:
We can have a discussion about whether we should change this, but this is not new behavior. Note that this is referring to the static options, so something like this would fail to compile:

```cc
  client.GetBucketMetadata("my-bucket",
     gcs::Generation(7) // compile-time error, buckets do not have generations
 );
```

while this would compile (and this is new behavior):

```cc
  client.GetBucketMetadata("my-bucket",
    g::Options{}.set<pubsub::RetryPolicyOption(...) // ignored
  );
```

Reply:
That's cool.  I was only thinking about the behavior for `Options`.